UserRelated/Main.py

```diff
@@ -0,0 +1,208 @@
+    :param shing_set: the user preferences shingle set
+    :param args: the given command-line arguments.
+    """
+    shing_set.add(TrackShape.LOOP.value) if args.shape == 1 else shing_set.add(TrackShape.CURVE.value)
+
+
+# Todo: this is a code duplication of the slopes_poc branch, we should have a separate class for LSH-ing we can reuse.
+def get_min_hash(shingles: set) -> MinHash:
+    """
+    given a set of shingles, creates a MinHash object updated with those shingles.
+    :param shingles: a set of track shingles.
+    :return: a MinHash object updated with the given shingles.
+    """
+    track_min_hash = MinHash(num_perm=128)
+    for shin in shingles:
+        track_min_hash.update(str(shin).encode('utf-8'))
+    return track_min_hash
+
+
+def create_user_shingles(args: argparse.Namespace) -> set:
+    """
+    Creates a set of shingles defining the user's path preferences.
+    :param args: the given command-line arguments.
+    :return: a set of shingles as described.
+    """
+    shing = set()
+    add_interest_points_shingles(shing, args)
+    add_length_shingle(shing, args)
+    add_difficulty_shingle(shing, args)
+    add_shape_shing(shing, args)
+    return shing
+
+
+def get_osm_tracks(area_path: str) -> dict:
+    """
+    Reads a dictionary with the data collected on paths in the given search area.
+    :param area_path: path to the file containing the data relevant to the given search are.
+    :return: a dictionary with data on tracks in the given area.
+    """
+    with open(area_path, 'r') as f:
+        osm_tracks_dict = json.load(f)
+    return osm_tracks_dict['tracks']
+
+
+def in_geo_limits(args: argparse.Namespace, track_data: dict) -> bool:
+    """
+    Checks if the given track is in the geographic limits the user had given.
+    :param args: command lines arguments.
+    :param track_data: a dictionary containing a track's data.
+    :return: true if the track lies in the geographic boundaries the user had set, false otherwise.
+    """
+    return (track_data['boundaries']['north'] <= args.north_lim and
+            track_data['boundaries']['south'] >= args.south_lim and
+            track_data['boundaries']['east'] <= args.east_lim and
+            track_data['boundaries']['west'] >= args.west_lim)
+
+
+if __name__ == '__main__':
+    """
+    Gets the user track preferences as command-line arguments, finds the most similar tracks to the request, and 
+    (in the future) presents the results to the user. The presentation of the results now is temporal and mainly 
+    used for testing.
+    Usage Example: baiersbronn 48.6 48.5 8.4 8.3 0 0 0 0 0 1 0 1 1 1 2
+    """
+    arg_parser = init_arg_parser()
+    command_line_args = arg_parser.parse_args()
+
+    user_shing = create_user_shingles(command_line_args)
+    lsh = MinHashLSH(threshold=SIMILARITY_THRESH, num_perm=128)
+    user_min_hash = get_min_hash(user_shing)
+    lsh.insert(USER_ID, user_min_hash)
```

Comment:
for what i see in the usage example here: http://ekzhu.com/datasketch/lsh.html, we shouldn't insert the one we're trying to query about. can you try removing this line (and line 206: unding the insert) and check if indeed we don't need to insert?

Reply:
I changed it and it worked, thanks!